### PR TITLE
[Reputation Oracle] Added pre-signup logic

### PR DESCRIPTION
**Pull request by eugenvoronov**
## Description 
Implemented an endpoint that returns [typed structured data](https://eips.ethereum.org/EIPS/eip-712) object that will be signed by user for sing up endpoint.

## Summary of changes
- Implemented pre-signup logic.
- Added web/pre-signup endpoint.
- Updated web3/signup to verify the signature on this generated message.

## How test the changes
`yarn test`
1. Call `/auth/web3/pre-signup` with:
```
{
  "chainId": 80001,
  "address": "0x0000000000000000000000000000000000000000"
}
```
2. Call `/auth/web3/signup` with:
```
{
  "chainId": 80001,
  "address": "0x0000000000000000000000000000000000000000",
  "signature": "string",
  "type": "OPERATOR"
}
```

## Related issues
[Reputation Oracle - Improve web3 sign up security #1568](https://github.com/humanprotocol/human-protocol/issues/1568)

### CODE REVIEW

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.dto.ts [hunk 1]
```diff
@@ -79,11 +86,34 @@ export class TokenCreateDto {
...
+
+  @ApiProperty()
+  @IsString()
+  public contents: string;
+}
+
+export class Web3PreSignUpDto {
+  @ApiProperty()
+  @IsEnum(ChainId)
+  public chainId: ChainId;
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/auth/auth.dto.ts:108]: chainId is not necessary

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -275,12 +279,30 @@ export class AuthService {
...
+
   public async web3Signup(data: Web3SignUpDto): Promise<AuthDto> {
-    const verified = await verifySignature(
-      WEB3_SIGNUP_MESSAGE,
-      data.signature,
-      [data.address],
-    );
+    const preSignUpData = this.prepareWeb3PreSignUpPayload(data);
+
+    const verified = await verifySignature(preSignUpData, data.signature, [
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:303]: I am getting Invalid signature when I test it. I am not sure why, but I think you might need to convert the preSignUpData to string

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts (hunk 2 above)
Comment (eugenvoronov) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:303]: Yes, you are right

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -282,19 +282,18 @@ export class AuthService {
   public prepareWeb3PreSignUpPayload(
     data: Web3PreSignUpDto,
   ): Web3PreSignUpPayloadDto {
-    const signer = this.web3Service.getSigner(data.chainId);
+    const privateKey = this.configService.get(ConfigNames.WEB3_PRIVATE_KEY);
```
[the remark this answers is not in the document]
Comment (eugenvoronov) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:285]: Agree

**Review by portuu3 — APPROVED** [review]

**Pull request merged**